### PR TITLE
Replace aws perl script with s3cmd WIP

### PR DESCRIPTION
**Pull request by joskar**
I'm not sure where to place the check for the s3cmd.
In the begging I thought `__detect_release_store_type` would be a good place but that would mean that every time someone will call `__detect_release_store_type` will make an ssh call to the remote system.  
Also I'm a bit confused why I need to have quotes here  https://github.com/joskar/edeliver/blob/s3/libexec/erlang#L696.  

I have tested the following pull request using these commands:

- mix edeliver build release
- mix edeliver deploy release production

### CODE REVIEW

File: libexec/erlang [hunk 1]
```diff
@@ -1157,6 +1145,20 @@ __show_upgrade_version_does_not_differ_error_message() {
     error_message "for automatic upgrades by setting AUTO_VERSION=revision in the config."
 }
 
+__find_remote_s3cmd() {
```
Comment (lessless) [inline libexec/erlang]: I would this is more of a check, then search. Maybe `__check_remote_s3cmd` or `__ensure_remote_s3_cmd` or  something alike.

File: libexec/erlang [hunk 2]
```diff
@@ -76,9 +76,22 @@ require_deploy_config() {
...
+    REQUIRED_CONFIGS+=("AWS_SECRET_ACCESS_KEY")
+  fi
+
   OPTIONAL_CONFIGS+=("VERSION")
 }
 
+s3_requirements() {
+  __detect_release_store_type
+  if [ "$RELEASE_STORE_TYPE" = "s3" ]; then
+    __find_remote_s3cmd
```
Comment (lessless) [inline libexec/erlang]: Are there any chances that this procedure will be used anywhere else? It doesn't look too big to inline to me.

**Comment by lessless** [comment]
I'm not a bash expert (sigh) but afaik quotes might be useful in case there is a spaces in the path.

File: libexec/erlang (hunk 1 above)
Comment (joskar) [reply libexec/erlang]: __ensure_remote_s3_cmd sounds good, will rename it

**Comment by joskar** [comment]
I forgot to point out. Other strategies have to be updated too.
I have only updated
- erlang-deploy-release
- erlang-build-release

File: libexec/erlang (hunk 2 above)
Comment (joskar) [reply libexec/erlang]: I created `s3_requirements` in order not to copy paste the same line in strategies over and over again. If you think that it shouldn't be a procedure I will inline it.

**Comment by lessless** [comment]
@joskar it will be really sad to have it done half-way. Are there any chances that you will have time and inspiration for that? 
No pressure though.

**Comment by joskar** [comment]
@lessless Yes I will finish it when I will have time :) . I started the pull request to get feedback from you because I'm not a bash expert XD and I'm worried if the things I'm doing are in the correct way.

**Comment by lessless** [comment]
All right, then Godspeed! :)

**Comment by bharendt** [comment]
Closing this as I'm clearing out stale branches. If anybody still wants this, please reopen.

**Pull request closed without merging**